teps/0096-pipelines-v1-release.md

```diff
@@ -0,0 +1,138 @@
+---
+status: proposed
+title: Pipelines V1 Release
+creation-date: '2021-11-29'
+last-updated: '2021-11-29'
+authors:
+- '@lbernick'
+- '@jerop'
+- '@bobcatfish'
+- '@vdemeester'
+see-also:
+- TEP-0033
+---
+
+# TEP-0096: Pipelines V1 Release
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Background](#background)
+- [Proposal](#proposal)
+  - [V1 Stability and Deprecation Policy](#v1-stability-and-deprecation-policy)
+  - [API Definition](#api-definition)
+  - [Use Cases](#use-cases)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+Today, Tekton provides Alpha and Beta APIs.
+This TEP proposes a path to releasing Tekton Pipelines V1 to provide a stable API that users can rely on.
+This TEP documents the API definition and stability policy for Pipelines V1,
+defines the use cases we would like to support for Pipelines V1, and discusses features needed to support these use cases.
+
+## Motivation
+
+To become the industry standard, cloud-native CI/CD platform,
+Tekton must provide users with a stable API they can rely on.
+Some users may be waiting for V1 stability before deciding to use Tekton.
+
+### Goals
+
+- Define stability policy for Pipelines V1.
+- Define supported and unsupported use cases for Pipelines V1.
+- Define criteria for a feature to be "required" or "optional" for Pipelines V1.
+- Define what updates, if any, are needed to the [Pipelines API definition](https://github.com/tektoncd/pipeline/blob/main/api_compatibility_policy.md#what-is-the-api) before V1 release.
+
+### Non Goals
+
+- Define the stability or V1 plans of other Tekton project.
+- Define stability policies for aspects of Pipelines unrelated to the API, such as performance, dogfooding,
+release automation, and testing and debugging features.
+- Discussion of software versioning. This TEP discusses only API versioning. 
+
+## Background
+
+Tekton currently uses Kubernetes' [API versioning guidelines](https://kubernetes.io/docs/reference/using-api/api-overview/#api-versioning)
+to define API stability levels, as described in the [compatibility policy](https://github.com/tektoncd/pipeline/blob/main/api_compatibility_policy.md#alpha-beta-and-ga).
+
+Tekton API components may contain features at a lower level of stability, gated behind feature flags,
+as described in [TEP-0033: Tekton Feature Gates](https://github.com/tektoncd/community/blob/main/teps/0033-tekton-feature-gates.md).
+
+Tekton follows Kubernetes' [deprecation policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/). 
+However, while Kubernetes defines stability periods in terms of months or releases, whichever is longer,
+Tekton defines stability periods only in terms of months. 
```

Comment:
This might change once we release V1, can it? The focus is months right now since Pipelines release are created every month. What is our plan to create release once we hit V1 API and/or V1 software versioning? We don't have to answer this here but I am just wondering while reading this since its being documented officially 🤣  unless I am missing the existing documentation.

Reply:
I think how often we want to create software releases is probably out of scope for the TEP, but we could choose to adopt Kubernetes language of "x months or x releases, whichever is longer". I don't see a strong reason in either direction; I think it's unlikely our release cadence would become significantly slower after a v1 release.